vaadin-radio-button-flow-demo/src/main/java/com/vaadin/flow/component/radiobutton/demo/RadioButtonGroupView.java

```diff
@@ -15,330 +15,287 @@
-                event -> valueInfo.setText(group.getValue()));
+        // source-example-heading: Required
+        Employee employee = new Employee();
+        Binder<Employee> binder = new Binder<>();
+
+        RadioButtonGroup<String> radioGroup = new RadioButtonGroup<>();
+        radioGroup.setLabel("Department");
+        radioGroup.setItems("Account Manager", "Designer", "Marketing Manager",
+                "Developer");
+        radioGroup.addThemeVariants(RadioGroupVariant.LUMO_VERTICAL);
+
+        binder.forField(radioGroup).asRequired("Please choose a department")
+                .bind(Employee::getTitle, Employee::setTitle);
+
+        Button button = new Button("Submit", event -> {
+            if (binder.writeBeanIfValid(employee)) {
+                Notification.show("Submit successful", 2000,
+                        Notification.Position.MIDDLE);
+            }
+        });
         // end-source-example
 
-        group.setId("button-group-read-only");
-        valueInfo.setId("selected-value-info");
-        button.setId("switch-read-only");
+        radioGroup.getStyle().set("margin-right", "5.5em");
+        HorizontalLayout layout = new HorizontalLayout(radioGroup, button);
+        layout.setAlignItems(FlexComponent.Alignment.BASELINE);
+        addCard("Validation", "Required", layout);
 
-        addCard("Read-only radio button group", group, button, valueInfo);
     }
 
-    private void addDisabledItems() {
-
-        Div valueInfo = new Div();
+    private void customOptions() {
         // begin-source-example
-        // source-example-heading: Radio button group with item enabled provider
-        RadioButtonGroup<String> group = new RadioButtonGroup<>();
-        group.setItems("foo", "bar", "baz");
-        group.setItemEnabledProvider(item -> !"bar".equals(item));
+        // source-example-heading: Customizing radio button label
+        RadioButtonGroup<Information> radioButton = new RadioButtonGroup<>();
+        radioButton.setLabel("User");
+        radioButton.setItems(
+                new Information("Gabriella",
+                        "https://randomuser.me/api/portraits/women/43.jpg"),
+                new Information("Rudi",
+                        "https://randomuser.me/api/portraits/men/77.jpg"),
+                new Information("Hamsa",
+                        "https://randomuser.me/api/portraits/men/35.jpg"),
+                new Information("Jacob",
+                        "https://randomuser.me/api/portraits/men/76.jpg"));
+        radioButton.addThemeVariants(RadioGroupVariant.LUMO_VERTICAL);
+
+        radioButton.setRenderer(new ComponentRenderer<>(information -> {
+            Div text = new Div();
+            text.setText(information.getText());
+
+            Image image = new Image();
+            image.setWidth("21px");
+            image.setHeight("21px");
+            image.setSrc(information.getImage());
+
+            FlexLayout wrapper = new FlexLayout();
```

Comment:
Wouldn't a horizontal layout with spacing work here? It's a higher-level abstraction so it's more convenient for a demo.

Reply:
Didn't change it because it looks ugly with HorizontalLayout.